AmplifyPlugins/API/AWSAPICategoryPlugin/Support/Utils/GraphQLResponseDecoder.swift

```diff
@@ -53,6 +53,11 @@ struct GraphQLResponseDecoder {
 
         case (.some(let data), .some(let errors)):
             do {
+                if data.count == 1, let first = data.first, case .null = first.value {
+                    let responseErrors = try decodeErrors(graphQLErrors: errors)
+                    return GraphQLResponse<R>.failure(.error(responseErrors))
+                }
```

Comment:
this was added due to graphql response that looks like
```
{
 "data": {
  "updatePost": null
 },
"errors": {
 ...
}
}
```

not too happy with this class, needs another pass on refactoring and better unit tests